### DISCUSSION TITLE
User Scenario: Getting Things Done

### DISCUSSION THREAD

**Discussion by bmulholland** [nvim-neorg/neorg#13] [Ideas]
I use a lightweight, GTD-inspired system to manage my tasks, and it would be great if neorg considered those scenarios as it's being built out. This is one way that neorg could be better than org mode, which has [a whole host of implementation approaches to consider](https://orgmode.org/worg/org-gtd-etc.html).

The most important parts of GTD are very simple:
* Have a central list of tasks
* If there's a goal to achieve that takes more than one task to achieve, it turns into a project.
* A project has a goal and a list of tasks
* On the central list of tasks, show all the non-project tasks *and* ONE task per project - the next one (aka "next action").

That's most of it. Otherwise it's useful to have some tagging and attributes per task and per project (e.g. "Work" vs "Personal"). There are other parts, like tasks you're waiting on others for, and events tied to a calendar, but I suspect you'd add those anyway...

**Comment by molleweide** [comment]
Some kind of tag system that u can make filters on would be dope as hell

**Comment by vhyrro** [comment]
This sounds like a great idea. Stuff like this is precisely planned for 0.2, where we will be remaking features like agenda, tangling etc. I'd imagine this stuff falls under the agenda category. I wonder what we'll name the agenda equivalent?

If it's alright I'll close this down for now so it doesn't sit here forever haha. I'll probably reopen this before starting to implement this stuff so we can discuss further then, k? If you have any extra ideas that randomly pop into your head then you can leave them here.

**Comment by bmulholland** [comment]
Yeah totally get why you'd want to close it out :) . Maybe topics like this would be a good use case for Github's Discussions?

**Reply by vhyrro** [reply]
It would be great for GitHub's discussions, which is precisely why this is now a discussion :P

**Comment by danymat** [comment]
What I love in gtd is the concept of inbox, a place to collect all actionable/non-actionable things in order to process them later.
I was thinking about an inbox command for Neorg in order to jot down a small text, that will be written in a inbox.norg file

**Reply by vhyrro** [reply]
That would definitely be awesome. I'm current working on a `core.ui` module that will allow us to quickly create nice looking prompts and floating windows. I hope to be able to use that to create quick popups like `New Note: ` etc.

EDIT: Inbetween everything else I've had to do I found some extra time to continue work on the UI module:

![ui test](https://user-images.githubusercontent.com/76052559/125997408-e4c2e51a-16c1-4c7f-9b2f-212602968808.png)

Very basic, but it works! That's always a good sign.

**Comment by danymat** [comment]
For those interested, I created a barebones module for gtd process: https://github.com/vhyrro/neorg/pull/60
